src/components/Chart/index.jsx

```diff
@@ -1,70 +1,131 @@
 import dayjs from 'dayjs';
+import isoWeek from 'dayjs/plugin/isoWeek';
 import PropTypes from 'prop-types';
 import { useCallback, useEffect, useState } from 'react';
+import { useMedia } from 'react-use';
 import { Area, AreaChart, Bar, ComposedChart, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts';
 import { useTheme } from 'styled-components';
 
 import { TIME_FILTER_OPTIONS } from '../../constants';
-import { formattedPercent } from '../../utils';
+import { formattedPercent, getWeekFormattedDate } from '../../utils';
 import CrosshairTooltip from './CrosshairTooltip';
 import Header from './Header';
 
+dayjs.extend(isoWeek);
+
+const getFilterLimitDate = (timeFilter) => {
+  let limitDate = new Date();
+
+  switch (timeFilter) {
+    case TIME_FILTER_OPTIONS.WEEK: {
+      limitDate.setDate(limitDate.getDate() - 7);
+      break;
+    }
+    case TIME_FILTER_OPTIONS.MONTH_1: {
+      limitDate.setMonth(limitDate.getMonth() - 1);
+      break;
+    }
+    case TIME_FILTER_OPTIONS.YEAR: {
+      limitDate.setFullYear(limitDate.getFullYear() - 1);
+      break;
+    }
+    default: {
+      limitDate.setDate(limitDate.getDate() - 7);
+      break;
+    }
+  }
+
+  return limitDate;
+};
+
+const getWeeklyAggregatedData = (limitDate, data) => {
+  const rawWeeklyAggregatedData = data
+    .filter((data) => new Date(data.time).getTime() > limitDate.getTime())
+    .reduce((previous, current) => {
+      const weekStart = dayjs(current.time).startOf('week');
+      const weekEnd = dayjs(current.time).endOf('week');
+
+      const weekId = `${dayjs(current.time).year()}-${weekStart.unix()}-${weekEnd.unix()}`;
+
+      return {
+        ...previous,
+        [weekId]: previous[weekId] ? previous[weekId] + current.value : current.value,
+      };
+    }, {});
+
+  let weeklyAggregatedData = [];
+
+  Object.keys(rawWeeklyAggregatedData).forEach((weekId) => {
+    const weekYear = weekId.split('-')[0];
+    const weekNumber = dayjs.unix(weekId.split('-')[1]).isoWeek() + 1;
+
+    weeklyAggregatedData.push({
+      time: dayjs().isoWeek(weekNumber).startOf('week').set('year', weekYear).format('YYYY-MM-DD'),
+      value: rawWeeklyAggregatedData[weekId],
+    });
+  });
+
+  return weeklyAggregatedData;
+};
+
 const Chart = ({ title, tooltipTitle, data, type, dataType, overridingActiveFilter, showTimeFilter }) => {
   const theme = useTheme();
   const [filteredData, setFilteredData] = useState(data);
   const [headerValue, setHeaderValue] = useState(null);
   const [activeDate, setActiveDate] = useState(null);
   const [activeFilter, setActiveFilter] = useState(TIME_FILTER_OPTIONS.MONTH_1);
   const [dailyChange, setDailyChange] = useState();
+  const [isWeeklyActive, setIsWeeklyActive] = useState(false);
+
+  const below500 = useMedia('(max-width: 500px)');
```

Comment:
I see this code used in a lot of places, might be better to share it?

Reply:
It can be removed from here, since the new `formatChartDate` will be used, but yes we may want to consider defining all the `belowxxx = useMedia(...)` in a single place and reuse them.